driver.go

```diff
@@ -161,43 +304,30 @@ func (c *conn) ExecContext(ctx context.Context, query string, args []driver.Name
 		return &result{rowsAffected: 0}, nil
 	}
 
-	if c.roTx != nil {
-		return nil, errors.New("cannot write in read-only transaction")
-	}
 	ss, err := prepareSpannerStmt(query, args)
 	if err != nil {
 		return nil, err
 	}
 
 	var rowsAffected int64
-	if c.rwTx == nil {
+	if c.tx == nil {
 		rowsAffected, err = c.execContextInNewRWTransaction(ctx, ss)
 	} else {
-		rowsAffected, err = c.rwTx.ExecContext(ctx, ss)
+		rowsAffected, err = c.tx.ExecContext(ctx, ss)
 	}
 	if err != nil {
 		return nil, err
 	}
 	return &result{rowsAffected: rowsAffected}, nil
 }
 
-func isDdl(query string) (bool, error) {
-
-	matchddl, err := regexp.MatchString(`(?is)^\n*\s*(CREATE|DROP|ALTER)\s+.+$`, query)
-	if err != nil {
-		return false, err
-	}
-
-	return matchddl, nil
-}
-
 func (c *conn) Close() error {
 	c.client.Close()
 	return nil
 }
 
 func (c *conn) Begin() (driver.Tx, error) {
-	panic("Using BeginTx instead")
+	return nil, fmt.Errorf("use BeginTx instead")
```

Comment:
Can we call `BeginTx` without a driver.TxOptions? E.g., 

```go
return c.BeginTx(context.Background(), driver.TxOptions{})
```

Reply:
Done